js/util.js

```diff
@@ -0,0 +1,46 @@
+const getRandomPositiveInteger = (a, b) => {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+getRandomPositiveInteger(1, 8);
+
+const getRandomPositiveFloat = (a, b, digits = 1) => {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return +result.toFixed(digits);
+};
+getRandomPositiveFloat(1.3, 4.9, 3);
+
+const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+
+const getRandomFeatures = (count, array) => {
+  const newFeatures = [];
+  if (count >= array.length) {
+    return array;
+  }
+  for (let i = 0; i < count; i++) {
+    let newElement = array[Math.floor(Math.random() * array.length)];
+    while (newFeatures.includes(newElement)) {
+      newElement = array[Math.floor(Math.random() * array.length)];
+    }
+    newFeatures.push(newElement);
+  }
+  return newFeatures;
```

Comment:
Более того ВСЮ эту функцию, теперь ты можешь заменить на свою собственную, используя существующий функционал makeGenerator:
```suggestion
const getRandomArrayElements = (count, array) => {
  const getArrayIndex = makeGenerator(count);
  return Array.from({length: count}, () => array[getArrayIndex()]);
};
```
Работать она будет точно так же, но используя алгоритм твоего кода

Reply:
Определение makeGenerator тогда просто повыше надо перенести, если захочешь заменить реализацию этой функции на новую предложенную